float64[16] EndPos